tcpclient/connection.go

```diff
@@ -71,3 +85,26 @@ func PrintGroupOfConnections(c []Connection) string {
 	return fmt.Sprintf("Total: %d, Dialing: %d, Established: %d, Closed: %d, Error: %d, NotInitiated: %d",
 		nTotal, nDialing, nEstablished, nClosed, nError, nNotInitiated)
 }
+
+func PrintFinalMetricsReport(c []Connection) string {
+	var avgToEstablished, minToEstablished, maxToEstablished time.Duration
+	var totalToEstalished time.Duration
+	for i, item := range c {
+		if i == 0 {
+			totalToEstalished = item.timeToEstablished()
+			minToEstablished = item.timeToEstablished()
+			maxToEstablished = item.timeToEstablished()
+		} else {
+			switch {
+			case item.timeToEstablished() < minToEstablished:
+				minToEstablished = item.timeToEstablished()
+			case item.timeToEstablished() > maxToEstablished:
+				maxToEstablished = item.timeToEstablished()
+			}
+			totalToEstalished += item.timeToEstablished()
+		}
+	}
+	avgToEstablished = totalToEstalished / time.Duration(len(c))
```

Comment:
i'd move the calcs outside of the function (one reports, another does the magic). Maybe, a function that, given an slice with connections, it gives you the stats

Reply:
👍